go/types/path.go

```diff
@@ -272,6 +276,7 @@ Switch:
 		for ; i < len(str); i++ {
 			c := str[i]
 			if c == '"' {
+				i++
```

Comment:
It looks like I missed writing a test like
```
test(`.foo["hello"`, "[ is missing closing ]")
```
I.e. a valid index (`"hello"`) but missing a closing `]`. Could you write that test? Seems like it would have been affected by this patch.